synapse/app/homeserver.py

```diff
@@ -185,6 +185,7 @@ def _listener_http(self, config, listener_config):
                         "/_matrix/client/r0": client_resource,
                         "/_matrix/client/unstable": client_resource,
                         "/_matrix/client/v2_alpha": client_resource,
+                        "/_matrix/versions": client_resource,
```

Comment:
Done

I was also unhappy to find this. In particular, the digging through when all I knew was that the thing 404'd, was not fun.